Bottom: 898ee2de1a88b99899117fea400f180d514a4f73
Top:    86ed09d7222c99944568a35ac1ea7717d47ba54e
Author: Christopher Voltz <cjunk@voltz.ws>
Date:   2016-04-29 02:49:49 -0500

Add tag, tag_col, tag_row keywords

Add new keywords:
  - tag(name) Associate the current row and column with name
  - tag_col(name) Return the column for the specified tag
  - tag_row(name) Return the row for the specified tag


---

diff --git a/script.js b/script.js
index 93024e1..2b98606 100644
--- a/script.js
+++ b/script.js
@@ -2,6 +2,7 @@ var tablecalc_table;
 var tablecalc_crow;
 var tablecalc_ccol;
 var tablecalc_labels = new Array();
+var tablecalc_tags = new Array();
 var tablecalc_defer = new Array();
 
 function tablecalcXY(st) {
@@ -95,6 +96,30 @@ function row() {
     return tablecalc_crow;
 }
 
+function tag(name) {
+    if (typeof tablecalc_tags[name] == "undefined") {
+        tablecalc_tags[name] = { row: row(), column: col() }
+    }
+
+    return "";
+}
+
+function tag_col(name) {
+    if (typeof tablecalc_tags[name] == "undefined") {
+        return 0;
+    }
+
+    return tablecalc_tags[name].column;
+}
+
+function tag_row(name) {
+    if (typeof tablecalc_tags[name] == "undefined") {
+        return 0;
+    }
+
+    return tablecalc_tags[name].row
+}
+
 function cell(x, y) {
     var tmp = tablecalcVal(x, y, tablecalc_table);
 
diff --git a/syntax.php b/syntax.php
index 9abd1ff..7307676 100644
--- a/syntax.php
+++ b/syntax.php
@@ -37,7 +37,7 @@ class syntax_plugin_tablecalc extends DokuWiki_Syntax_Plugin {
 
         $signs = "-~=+*.,;\/!|&\(\)";
         $pattern = "/[$signs]*([a-zA-Z]+)\(/is";
-        $aAllowed = array("cell", "row", "col", "sum", "average", "count", "nop", "round", "range", "label", "min", "max", "calc", "check", "compare");
+        $aAllowed = array("cell", "row", "col", "sum", "average", "count", "nop", "round", "range", "label", "min", "max", "calc", "check", "compare", "tag", "tag_row", "tag_col");
 
         if (preg_match_all($pattern, $match, $aMatches)) {
             foreach ($aMatches[1] as $f) {
